examples/guidelines.md

```diff
@@ -0,0 +1,78 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Example Guidelines
+
+## An Example Is
+
+An example demonstrates running a real application/framework/workload
+on Kubernetes in a meaningful way. It is educational and informative.
+
+Examples are not:
+
+* Full app deployments, ready to use, with no explanation. These
+  belong either
+  [here](https://github.com/kubernetes/deployment-manager) or in
+  something like [Helm](https://helm.sh/).
+* Simple toys to show how to use a Kubernetes feature. These belong in
+  the [user guide](../docs/user-guide/).
+* Demos that follow a script to show a Kubernetes feature in
+  action. Example: killing a node to demonstrate controller
+  self-healing.
+
+## An Example Includes
+
+### Up front
+
+* Should have a "this is what you'll learn" section.
+* Optionally, if the example is long: Should have a section that brings up the app in the fewest number of commands (TL;DR / quickstart), without cloning the repo (kubectl -f http:...).
+* Should point to documentation of prerequisites.
+  * Create a cluster (e.g., single-node docker).
+  * Setup kubectl.
+* Should specify which release of Kubernetes is required and any other prerequisites, such as DNS, a cloudprovider with PV provisioning, a cloudprovider with external load balancers, etc.
+  * Point to general documentation about alternatives for those mechanisms rather than present the alternatives in each example.
+
+### Throughout
+
+* Should point to documentation on first mention: pods, kubectl, etc.
+* Most examples should be cloudprovider-independent (e.g., using PVCs, not PDs).
+  * Other examples with cloudprovider-specific bits could be somewhere else.
+* Actually show the app working -- output, ascii animations, or screenshots.
+* Follows [config best practices](../docs/user-guide/config-best-practices.md).
```

Comment:
We probably need to make an exception for use of namespaces, since namespace creation would add an extra step to every example. We should have one example that shows that. Maybe convert the blog post to an example:
http://blog.kubernetes.io/2015/08/using-kubernetes-namespaces-to-manage.html